android/src/main/java/com/fidelreactlibrary/adapters/ResultsAdapter.java

```diff
@@ -126,6 +126,8 @@ public WritableMap getAdaptedObjectFor(Object data) {
     private String getJSErrorType(FidelErrorType errorType) {
         if (errorType == FidelErrorType.UserCanceled.INSTANCE) {
             return "userCanceled";
+        } else if (errorType == FidelErrorType.DeviceNotSecure.INSTANCE) {
+            return "deviceNotSecure";
```

Comment:
In case this is the error that is returned to the App using the SDK, should we be more specific on why isn't the device secure? or it isn't a best practice to be transparent on that kind of stuff?

Reply:
> 

There are a couple of reasons I considered making this error not so specific/transparent:
1. Right now we check only if the device is rooted/jailbroken. In the future we might add more checks related to the security of the device and we would not need to change the error.
2. I thought it is best to not use words like "rooted" or "jailbroken" in our communication of this error, as that would give reverse engineering clues to an attacker.